### ISSUE TITLE
Icinga2 crashes during notification when plugin output contains a $

### ISSUE THREAD

**Issue by Mikesch-mp** [Icinga/icinga2#9186]
## Describe the bug

Icinga2 crashes during notification when plugin output contains a $

## To Reproduce

1. Setup a windows agent and dont install the powershell framework and plugins
2. Setup a normal windows service check that uses the powershell famework & plugins

## Expected behavior

Icinga2 should escape all the plugin output or not treat any text from pluginoutput as macro.

## Log
```
[2022-01-20 21:22:26 +0100] information/Checkable: Checkable 'myvm01!Windows Service VMTools' has 1 notification(s). Checking filters for type 'Problem', sends will be logged.
[2022-01-20 21:22:26 +0100] information/Notification: Sending 'Problem' notification 'vmyvm01!Windows Service VMTools!jira-service' for user 'jira-monitoring'
[2022-01-20 21:22:26 +0100] warning/PluginUtility: Error: Closing $ not found in macro format string.

Stacktrace:
 0# __cxa_throw in /usr/lib64/icinga2/sbin/icinga2
 1# void boost::throw_exception<boost::exception_detail::error_info_injector<std::runtime_error> >(boost::exception_detail::error_info_injector<std::runtime_error> const&) in /usr/lib64/icinga2/sbin/icinga2
 2# void boost::exception_detail::throw_exception_<std::runtime_error>(std::runtime_error const&, char const*, char const*, int) in /usr/lib64/icinga2/sbin/icinga2
 3# icinga::MacroProcessor::InternalResolveMacros(icinga::String const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::String*, std::function<icinga::Value (icinga::Value const&)> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 4# icinga::MacroProcessor::InternalResolveMacros(icinga::String const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::String*, std::function<icinga::Value (icinga::Value const&)> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 5# icinga::MacroProcessor::ResolveMacros(icinga::Value const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::String*, std::function<icinga::Value (icinga::Value const&)> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 6# icinga::MacroProcessor::ResolveArguments(icinga::Value const&, boost::intrusive_ptr<icinga::Dictionary> const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 7# icinga::PluginUtility::ExecuteCommand(boost::intrusive_ptr<icinga::Command> const&, boost::intrusive_ptr<icinga::Checkable> const&, boost::intrusive_ptr<icinga::CheckResult> const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int, std::function<void (icinga::Value const&, icinga::ProcessResult const&)> const&) in /usr/lib64/icinga2/sbin/icinga2
 8# icinga::PluginNotificationTask::ScriptFunc(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool) in /usr/lib64/icinga2/sbin/icinga2
 9# std::_Function_handler<icinga::Value (std::vector<icinga::Value, std::allocator<icinga::Value> > const&), std::enable_if<std::is_function<std::remove_pointer<void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool)>::type>::value&&(!std::is_same<void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool), icinga::Value (*)(std::vector<icinga::Value, std::allocator<icinga::Value> > const&)>::value), std::function<icinga::Value (std::vector<icinga::Value, std::allocator<icinga::Value> > const&)> >::type icinga::WrapFunction<void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool)>(void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool))::{lambda(std::vector<icinga::Value, std::allocator<icinga::Value> > const&)#1}>::_M_invoke(std::_Any_data const&, std::vector<icinga::Value, std::allocator<icinga::Value> > const&) in /usr/lib64/icinga2/sbin/icinga2
10# icinga::Function::Invoke(std::vector<icinga::Value, std::allocator<icinga::Value> > const&) in /usr/lib64/icinga2/sbin/icinga2
11# icinga::NotificationCommand::Execute(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::NotificationType const&, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool) in /usr/lib64/icinga2/sbin/icinga2
12# icinga::Notification::ExecuteNotificationHelper(icinga::NotificationType, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, bool, icinga::String const&, icinga::String const&) in /usr/lib64/icinga2/sbin/icinga2
13# 0x0000000000CCD0EB in /usr/lib64/icinga2/sbin/icinga2
14# bool icinga::ThreadPool::Post<std::function<void ()> >(std::function<void ()>, icinga::SchedulerPolicy)::{lambda()#1}::operator()() const in /usr/lib64/icinga2/sbin/icinga2
15# boost::asio::detail::executor_op<boost::asio::detail::work_dispatcher<bool icinga::ThreadPool::Post<std::function<void ()> >(std::function<void ()>, icinga::SchedulerPolicy)::{lambda()#1}>, std::allocator<void>, boost::asio::detail::scheduler_operation>::do_complete(void*, std::allocator<void>*, boost::system::error_code const&, unsigned long) in /usr/lib64/icinga2/sbin/icinga2
16# boost::asio::detail::scheduler::run(boost::system::error_code&) in /usr/lib64/icinga2/sbin/icinga2
17# boost::asio::detail::posix_thread::func<boost::asio::thread_pool::thread_function>::run() in /usr/lib64/icinga2/sbin/icinga2
18# boost_asio_detail_posix_thread_function in /usr/lib64/icinga2/sbin/icinga2
19# 0x00007F572BF05EA5 in /lib64/libpthread.so.0
20# clone in /lib64/libc.so.6

Context:

        (0) Resolving macros for string 'Command C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -C "try { Use-Icinga; } catch { Write-Output 'The Icinga PowerShell Framework is either not installed on the system or not configured properly. Please check https://icinga.com/docs/windows for further details'; exit 3; }; Exit-IcingaPluginNotInstalled 'Invoke-IcingaCheckService'; exit Invoke-IcingaCheckService" -Service VMTools -Exclude $null -Status Running -NoPerfData failed to execute: 19, "Das Medium ist schreibgesch�tzt."'
        (1) Resolving macros for string '$jira_description$ \\ Time of Event: $jira_laststatechange$'
[2022-01-20 21:22:26 +0100] warning/PluginNotificationTask: Notification command for object 'myvm01!Windows Service VMTools' (PID: -1, arguments: 'sh' '-c' '') terminated with exit code 3, output: Error: Closing $ not found in macro format string.

Stacktrace:
 0# __cxa_throw in /usr/lib64/icinga2/sbin/icinga2
 1# void boost::throw_exception<boost::exception_detail::error_info_injector<std::runtime_error> >(boost::exception_detail::error_info_injector<std::runtime_error> const&) in /usr/lib64/icinga2/sbin/icinga2
 2# void boost::exception_detail::throw_exception_<std::runtime_error>(std::runtime_error const&, char const*, char const*, int) in /usr/lib64/icinga2/sbin/icinga2
 3# icinga::MacroProcessor::InternalResolveMacros(icinga::String const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::String*, std::function<icinga::Value (icinga::Value const&)> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 4# icinga::MacroProcessor::InternalResolveMacros(icinga::String const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::String*, std::function<icinga::Value (icinga::Value const&)> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 5# icinga::MacroProcessor::ResolveMacros(icinga::Value const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::String*, std::function<icinga::Value (icinga::Value const&)> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 6# icinga::MacroProcessor::ResolveArguments(icinga::Value const&, boost::intrusive_ptr<icinga::Dictionary> const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::CheckResult> const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int) in /usr/lib64/icinga2/sbin/icinga2
 7# icinga::PluginUtility::ExecuteCommand(boost::intrusive_ptr<icinga::Command> const&, boost::intrusive_ptr<icinga::Checkable> const&, boost::intrusive_ptr<icinga::CheckResult> const&, std::vector<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> >, std::allocator<std::pair<icinga::String, boost::intrusive_ptr<icinga::Object> > > > const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool, int, std::function<void (icinga::Value const&, icinga::ProcessResult const&)> const&) in /usr/lib64/icinga2/sbin/icinga2
 8# icinga::PluginNotificationTask::ScriptFunc(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool) in /usr/lib64/icinga2/sbin/icinga2
 9# std::_Function_handler<icinga::Value (std::vector<icinga::Value, std::allocator<icinga::Value> > const&), std::enable_if<std::is_function<std::remove_pointer<void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool)>::type>::value&&(!std::is_same<void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool), icinga::Value (*)(std::vector<icinga::Value, std::allocator<icinga::Value> > const&)>::value), std::function<icinga::Value (std::vector<icinga::Value, std::allocator<icinga::Value> > const&)> >::type icinga::WrapFunction<void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool)>(void (*)(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, int, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool))::{lambda(std::vector<icinga::Value, std::allocator<icinga::Value> > const&)#1}>::_M_invoke(std::_Any_data const&, std::vector<icinga::Value, std::allocator<icinga::Value> > const&) in /usr/lib64/icinga2/sbin/icinga2
10# icinga::Function::Invoke(std::vector<icinga::Value, std::allocator<icinga::Value> > const&) in /usr/lib64/icinga2/sbin/icinga2
11# icinga::NotificationCommand::Execute(boost::intrusive_ptr<icinga::Notification> const&, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, icinga::NotificationType const&, icinga::String const&, icinga::String const&, boost::intrusive_ptr<icinga::Dictionary> const&, bool) in /usr/lib64/icinga2/sbin/icinga2
12# icinga::Notification::ExecuteNotificationHelper(icinga::NotificationType, boost::intrusive_ptr<icinga::User> const&, boost::intrusive_ptr<icinga::CheckResult> const&, bool, icinga::String const&, icinga::String const&) in /usr/lib64/icinga2/sbin/icinga2
13# 0x0000000000CCD0EB in /usr/lib64/icinga2/sbin/icinga2
14# bool icinga::ThreadPool::Post<std::function<void ()> >(std::function<void ()>, icinga::SchedulerPolicy)::{lambda()#1}::operator()() const in /usr/lib64/icinga2/sbin/icinga2
15# boost::asio::detail::executor_op<boost::asio::detail::work_dispatcher<bool icinga::ThreadPool::Post<std::function<void ()> >(std::function<void ()>, icinga::SchedulerPolicy)::{lambda()#1}>, std::allocator<void>, boost::asio::detail::scheduler_operation>::do_complete(void*, std::allocator<void>*, boost::system::error_code const&, unsigned long) in /usr/lib64/icinga2/sbin/icinga2
16# boost::asio::detail::scheduler::run(boost::system::error_code&) in /usr/lib64/icinga2/sbin/icinga2
17# boost::asio::detail::posix_thread::func<boost::asio::thread_pool::thread_function>::run() in /usr/lib64/icinga2/sbin/icinga2
18# boost_asio_detail_posix_thread_function in /usr/lib64/icinga2/sbin/icinga2
19# 0x00007F572BF05EA5 in /lib64/libpthread.so.0
20# clone in /lib64/libc.so.6

Context:

        (0) Resolving macros for string 'Command C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -C "try { Use-Icinga; } catch { Write-Output 'The Icinga PowerShell Framework is either not installed on the system or not configured properly. Please check https://icinga.com/docs/windows for further details'; exit 3; }; Exit-IcingaPluginNotInstalled 'Invoke-IcingaCheckService'; exit Invoke-IcingaCheckService" -Service VMTools -Exclude $null -Status Running -NoPerfData failed to execute: 19, "Das Medium ist schreibgesch�tzt."'
        (1) Resolving macros for string '$jira_description$ \\ Time of Event: $jira_laststatechange$'
[2022-01-20 21:22:26 +0100] information/Notification: Completed sending 'Problem' notification 'myvm01!Windows Service VMTools!jira-service' for checkable 'myvm01!Windows Service VMTools' and user 'jira-monitoring' using command 'JIRA Service Notification'.

```
## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`): 2.13.2
* Operating System and version: Centos 7.9.2009
* Enabled features (`icinga2 feature list`): api checker ido-mysql influxdb mainlog notification
* Icinga Web 2 version and modules (System - About): 2.95
* Config validation (`icinga2 daemon -C`):
```
chief01:/var/log/icinga2 $ icinga2 daemon -C
[2022-01-21 08:27:58 +0100] information/cli: Icinga application loader (version: 2.13.2-1)
[2022-01-21 08:27:58 +0100] information/cli: Loading configuration file(s).
[2022-01-21 08:27:59 +0100] information/ConfigItem: Committing config item(s).
[2022-01-21 08:27:59 +0100] information/ApiListener: My API identity: chief01
[2022-01-21 08:28:09 +0100] information/WorkQueue: #4 (DaemonUtility::LoadConfigFiles) items: 0, rate: 48.4/s (2904/min 2904/5min 2904/15min);
[2022-01-21 08:28:09 +0100] information/WorkQueue: #5 (InfluxdbWriter, influxdb) items: 0, rate:  0/s (0/min 0/5min 0/15min);
[2022-01-21 08:28:09 +0100] information/WorkQueue: #7 (ApiListener, RelayQueue) items: 0, rate:  0/s (0/min 0/5min 0/15min);
[2022-01-21 08:28:09 +0100] information/WorkQueue: #8 (ApiListener, SyncQueue) items: 0, rate:  0/s (0/min 0/5min 0/15min);
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 InfluxdbWriter.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 NotificationComponent.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 IdoMysqlConnection.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 CheckerComponent.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 3 Users.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 ServiceGroup.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 5 TimePeriods.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 16405 Services.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 482 Zones.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 399 ScheduledDowntimes.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 6 NotificationCommands.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 4 HostGroups.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 16654 Notifications.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 616 Downtimes.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 28322 Dependencies.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 IcingaApplication.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1876 Hosts.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 EventCommand.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 483 Endpoints.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 155 Comments.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 FileLogger.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 16 ApiUsers.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 311 CheckCommands.
[2022-01-21 08:28:11 +0100] information/ConfigItem: Instantiated 1 ApiListener.
[2022-01-21 08:28:11 +0100] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2022-01-21 08:28:11 +0100] information/cli: Finished validating the configuration file(s).
```
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes.

**Comment by Al2Klimov** [comment]
I'm pretty sure Icinga 2 doesn’t interpret plugin output like this.

```
[2023-05-03 09:55:31 +0200] warning/PluginCheckTask: Check command for object 'dummy!9186' (PID: 92842, arguments: 'python3' '-c' 'print("\x24host.name\x24"); exit(128)') terminated with exit code 128, output: $host.name$
```

**Comment by Mikesch-mp** [comment]
Lol? Not even try to test it like i wrote? Icinga going down the road faster and faster...

**Comment by Al2Klimov** [comment]
Even if I test it like you suggested,

> * Setup a windows agent and dont install the powershell framework and plugins
> * Setup a normal windows service check that uses the powershell famework & plugins

Icinga can successfully _invoke_ the notification script:

```
[2023-05-03 10:19:04 +0000] warning/PluginNotificationTask: Notification command for object 'desktop-1orn81m!c. p. u.' (PID: 190, arguments: '/etc/icinga2/scripts/mail-service-notification.sh' '-4' '10.27.0.157' '-6' '' '-b' '' '-c' '' '-d' '2023-05-03 10:19:04 +0000' '-e' 'c. p. u.' '-l' 'desktop-1orn81m' '-n' 'desktop-1orn81m' '-o' 'The Icinga PowerShell Framework is either not installed on the system or not configured properly. Please check https://icinga.com/docs/windows for further details

Error:

Die Benennung "Use-Icinga" wurde nicht als Name eines Cmdlet, einer Funktion, einer Skriptdatei oder eines ausführbaren Programms erkannt. Überprüfen Sie die Schreibweise des Namens, oder ob der Pfad korrekt ist (sofern enthalten), und wiederholen Sie den Vorgang.

Components:



Module-Path:

C:\WINDOWS\ServiceProfiles\NetworkService\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules' '-r' 'aklimov@netways.de' '-s' 'UNKNOWN' '-t' 'PROBLEM' '-u' 'c. p. u.') terminated with exit code 36, output: mail: cannot send message: Process exited with a non-zero status
```

**Comment by Al2Klimov** [comment]
To make the test case perfect:

```
[2023-05-03 11:07:36 +0000] warning/PluginNotificationTask: Notification command for object 'desktop-1orn81m!c. p. u.' (PID: 157, arguments: '/etc/icinga2/scripts/mail-service-notification.sh' '-4' '10.27.0.157' '-6' '' '-b' '' '-c' '' '-d' '2023-05-03 11:07:36 +0000' '-e' 'c. p. u.' '-l' 'desktop-1orn81m' '-n' 'desktop-1orn81m' '-o' 'The Icinga PowerShell Framework is either not installed on the system or not configured properly. Please check https://icinga.com/docs/windows for further details

Error:

Die Benennung "Use-Icinga" wurde nicht als Name eines Cmdlet, einer Funktion, einer Skriptdatei oder eines ausführbaren Programms erkannt. Überprüfen Sie die Schreibweise des Namens, oder ob der Pfad korrekt ist (sofern enthalten), und wiederholen Sie den Vorgang.

Components:



Single-Dollar-Sign: $ Module-Path:

C:\WINDOWS\ServiceProfiles\NetworkService\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules' '-r' 'aklimov@netways.de' '-s' 'UNKNOWN' '-t' 'PROBLEM' '-u' 'c. p. u.') terminated with exit code 36, output: mail: cannot send message: Process exited with a non-zero status
```

Search for _Single-Dollar-Sign_, next to it is a -surprise, surprise- single dollar sign. This is part of the PS missing-plugin output Icinga successfully passes to the notification command.

### TL;DR

Works for me.

@julianbrost If you agree with me, please close this.
